src/ledger.ts

```diff
@@ -1538,7 +1557,81 @@ export class LedgerConfirmMultisigAddress extends LedgerBitcoinV2WithRegistratio
       // run the app version support checks, register wallet if necessary
       await super.run();
       await this.registerWallet();
-      return this.getAddress();
+      // TODO doesn't handle catching error where policy doesn't match well
+      return await this.getAddress();
+    } finally {
+      super.closeTransport();
+    }
+  }
+}
+
+interface LedgerV2SignTxConstructorArguments
+  extends RegistrationConstructorArguments {
+  psbt: string | Buffer;
+
+  progressCallback?: () => void;
+}
+
+type InputIndex = number;
+// a Buffer with either a 33-byte compressed pubkey or a 32-byte
+// x-only pubkey whose corresponding secret key was used to sign;
+type PubKey = Buffer;
+// a Buffer with the corresponding signature.
+type SignatureBuffer = Buffer;
+// return type of ledger after signing
+type LedgerSignatures = [InputIndex, PubKey, SignatureBuffer];
+export class LedgerV2SignMultisigTransaction extends LedgerBitcoinV2WithRegistrationInteraction {
+  private psbt: PsbtV2;
+
+  // optionally, a callback that will be called every time a signature is produced during
+  //  * the signing process. The callback does not receive any argument, but can be used to track progress.
+  public progressCallback?: () => void;
+
+  private signatures: LedgerSignatures[] = [];
+
+  constructor({
+    psbt,
+    progressCallback,
+    name,
+    braid,
+    policyHmac,
+  }: LedgerV2SignTxConstructorArguments) {
+    super({ name, braid, policyHmac });
+
+    if (progressCallback) this.progressCallback = progressCallback;
+
+    const psbtVersion = getPsbtVersionNumber(psbt);
+    if (psbtVersion === 0) {
+      this.psbt = PsbtV2.FromV0(psbt);
+    } else if (psbtVersion === 2) {
+      this.psbt = new PsbtV2(psbt);
+    } else {
+      throw new Error(`PSBT of unsupported version ${psbtVersion}`);
+    }
+  }
+
+  async signPsbt(): Promise<LedgerSignatures[]> {
+    return this.withApp(async (app: AppClient) => {
+      this.signatures = await app.signPsbt(
+        this.psbt,
+        this.walletPolicy.toLedgerPolicy(),
+        this.policyHmac || null,
+        this.progressCallback
```

Comment:
is it possible for this to be undefined? is it OK if it is?

Reply:
Yeah. Probably won't be used most of the time. The use case (which will be REALLY nice if we can leverage) is for a progress tracker. This gets called anytime an input is signed. But in most cases it will be undefined and not used.